### PR TITLE
chore: update naming and use crs-version where needed

### PR DESCRIPTION
**Pull request by fzipi**
- use crs-version parameter to generate documentation
- change naming from `owasp-modsecurity-crs` to `coreruleset`

### CODE REVIEW

**Comment by dune73** [comment]
Why do you do a folder `coreruleset` instead of `crs`?

**Comment by fzipi** [comment]
That's how we provide our tarballs. If we want to rename everything, we can, but not before tomorrow, right?

We need to rename our main repo from `coreruleset/coreruleset` to `coreruleset/crs`, basically. And change everything around (docs, wiki, installer, links).

**Comment by dune73** [comment]
No, that's not what I am talking about. I'm talking about

```
tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/coreruleset
```

If you do this to mirror the repo name, then cool. But it's obviously an arbitrary choice (and better than `owasp-modsecurity-crs`.

File: content/deployment/quick_start.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ Note that while it's common practice to make a new `modsecurity.d` folder, as ou
 
 ```bash
 mkdir /etc/httpd/modsecurity.d
-tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/owasp-modsecurity-crs
+tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/coreruleset
```
Comment (fzipi) [inline content/deployment/quick_start.md:82]:
```suggestion
tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/crs
```
It's easier if you add a suggestion in the code itself. Less ambiguous, easy to follow.

File: content/deployment/quick_start.md [hunk 2]
```diff
@@ -95,7 +95,7 @@ In many scenarios, the default example CRS configuration will be a good enough s
 Once any settings have been changed within the example configuration file, as needed, it should be renamed to remove the .example portion, like so:
 
 ```bash
-cd /etc/httpd/modsecurity.d/owasp-modsecurity-crs/
+cd /etc/httpd/modsecurity.d/coreruleset/
```
Comment (fzipi) [inline content/deployment/quick_start.md:98]:
```suggestion
cd /etc/httpd/modsecurity.d/crs/
```

File: content/deployment/quick_start.md [hunk 3]
```diff
@@ -104,8 +104,8 @@ mv crs-setup.conf.example crs-setup.conf
 The last step is to tell the web server where the rules are. This is achieved by `include`-ing the rule configuration files in the `httpd.conf` file. Again, this example demonstrates using Apache, but the process is similar on other systems (see the [extended install]({{< ref "install.md" >}}) page for details).
 
 ```bash
-echo 'IncludeOptional /etc/httpd/owasp-modsecurity-crs/crs-setup.conf' >> /etc/httpd/conf/httpd.conf
-echo 'IncludeOptional /etc/httpd/owasp-modsecurity-crs/rules/*.conf' >> /etc/httpd/conf/httpd.conf
+echo 'IncludeOptional /etc/httpd/coreruleset/crs-setup.conf' >> /etc/httpd/conf/httpd.conf
+echo 'IncludeOptional /etc/httpd/coreruleset/rules/*.conf' >> /etc/httpd/conf/httpd.conf
```
Comment (fzipi) [inline content/deployment/quick_start.md:108]:
```suggestion
echo 'IncludeOptional /etc/httpd/crs/crs-setup.conf' >> /etc/httpd/conf/httpd.conf
echo 'IncludeOptional /etc/httpd/crs/rules/*.conf' >> /etc/httpd/conf/httpd.conf
```

**Comment by fzipi** [comment]
What about now?

**Comment by dune73** [comment]
Very good thanks. 

What I see still missing is the plugins includes. Going to comment in code.

**Comment by dune73** [comment]
Ah, no, need to comment here, since you did not touch those lines:

```
This would be:

```

Include modsecurity.d/crs/crs-setup.conf

Include modsecurity.d/crs/plugins/*-config.conf
Include modsecurity.d/crs/plugins/*-before.conf

Include modsecurity.d/crs/rules/*.conf

Include modsecurity.d/crs/plugins/*-after.conf


```

**Comment by fzipi** [comment]
ping @dune73

**Comment by dune73** [comment]
Sorry, I do not get the question. What's with nginx? Same layout, I think.

**Comment by fzipi** [comment]
So include works the same way in nginx?

**Comment by dune73** [comment]
@airween could you please confirm?

**Comment by dune73** [comment]
There is a problem with the tar now:

```
tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/crs

```

We're likely ending up with something along `modsecurity.d/crs/coreruleset-nightly`.

Not sure how we want to deal with this.

**Comment by dune73** [comment]
The extended install lacks the extraction step.

I suggest to fix the tar above and then to copy that over to the extended install.

File: config/_default/config.toml [hunk 4]
```diff
@@ -43,6 +43,7 @@ home = [ "HTML", "RSS", "JSON"]
   # CRS related parameters
   crs_latest_release = "3.3.4"
```
Comment (dune73) [inline config/_default/config.toml:44]: This better be 4.0.0, I guess.

**Comment by dune73** [comment]
Please update `gpg --verify coreruleset-3.3.2.tar.gz.asc v3.3.2.tar.gz` to v4. (can't comment in the file since you did not touch this).

File: content/deployment/quick_start.md [hunk 5]
```diff
@@ -78,10 +78,12 @@ The examples presented below demonstrate using Apache. For information on config
 Note that while it's common practice to make a new `modsecurity.d` folder, as outlined below, this isn't strictly necessary. The path scheme outlined is common on RHEL-based operating systems; the Apache path used may need to be adjusted to match the server's installation.
 
 ```bash
-mkdir /etc/httpd/modsecurity.d
-tar -zxvf v{{< param crs_latest_release >}}.tar.gz -C /etc/httpd/modsecurity.d/owasp-modsecurity-crs
+mkdir {{< param crs_install_dir >}}
+tar -xzvf v{{< param crs_latest_release >}}.tar.gz --strip-components 1 -C {{< param crs_install_dir >}}
```
Comment (dune73) [inline content/deployment/quick_start.md:82]: works!

**Review by dune73 — CHANGES_REQUESTED** [review]

File: config/_default/config.toml (hunk 4 above)
Comment (fzipi) [reply config/_default/config.toml:44]: The build failed to I've reverted this change. Will push it after release.

**Comment by airween** [comment]
> Ah, no, need to comment here, since you did not touch those lines:
> 
> ```
> 
> Include modsecurity.d/crs/crs-setup.conf
> 
> Include modsecurity.d/crs/plugins/*-config.conf
> Include modsecurity.d/crs/plugins/*-before.conf
> 
> Include modsecurity.d/crs/rules/*.conf
> 
> Include modsecurity.d/crs/plugins/*-after.conf
> ```

This works for me with Nginx (with different directory name) - but everything is the same.

```
Include /PATH/TO/CRS/crs-setup.conf

Include /PATH/TO/CRS/plugins/*-config.conf
Include /PATH/TO/CRS/plugins/*-before.conf

Include /PATH/TO/CRS/rules/*.conf

Include /PATH/TO/CRS/plugins/*-after.conf
```

**Comment by fzipi** [comment]
Then this `Include` simplifies everything

**Comment by airween** [comment]
> Then this `Include` simplifies everything. But there is no `IncludeOptional`... right? So it may fail if you don't have plugins.

Yes, Nginx does not support `IncludeOptional` at all.

**Comment by dune73** [comment]
That's why we ship with

```
dune73@leander plugins>l
total 12K
drwxr-xr-x  2 dune73 dune73 4.0K Nov 23 09:38 .
drwxr-xr-x 10 dune73 dune73 4.0K Feb 14 14:41 ..
-rw-r--r--  1 dune73 dune73    0 Nov 23 09:38 empty-after.conf
-rw-r--r--  1 dune73 dune73    0 Nov 23 09:38 empty-before.conf
-rw-r--r--  1 dune73 dune73    0 Nov 23 09:38 empty-config.conf

```

**Review by dune73 — APPROVED** [review]

**Pull request merged**